src/services/SocketService.ts

```diff
@@ -39,19 +39,33 @@ export class SocketService {
     });
   }
 
-  public joinRoomAndWaitForEvent(roomId: string): Promise<any> {
+  public joinRoomAndWaitForEvent(
+    roomId: string,
+    sourceChain: string,
+    destinationChain: string,
+    destinationAddress: string
+  ): Promise<any> {
     return new Promise(async (resolve) => {
       await this.createSocket();
       const ms = 1.8e6; //30 minutes
       const timeout = setTimeout(() => {
+        this.socket.off("bridge-event");
         this.disconnect();
       }, ms);
-      this.socket.emit("room:join", roomId, () => {
-        this.socket.on("bridge-event", (data: any) => {
+      this.socket.emit("room:join", roomId);
+      this.socket.on("bridge-event", (data: any) => {
+        const attributes = data.Attributes;
+        const sourceChainMatch = attributes.sourceChain.toLowerCase() === sourceChain.toLowerCase();
+        const destChainMatch =
+          attributes.destinationChain.toLowerCase() === destinationChain.toLowerCase();
+        const destAddressMatch = attributes.destinationAddress === destinationAddress;
```

Comment:
do we need to lower case this one too?

Reply:
i think so, because users can enter chains all caps or all lowercase. But the events from microservices are returning capitalized chains

Reply:
oh, i meant the `destinationAddress` 🙂